docs/apps/interchain-accounts/legacy/keeper-api.md

```diff
@@ -0,0 +1,121 @@
+<!--
+order: 3
+-->
+
+# Keeper API
+
+## Deprecation Notice
+
+**This document is deprecated and will be removed in future releases**.
+
+The controller submodule keeper exposes two legacy functions that allow respectively for custom authentication modules to register interchain accounts and send packets to the interchain account.
+
+## `RegisterInterchainAccount`
+
+The authentication module can begin registering interchain accounts by calling `RegisterInterchainAccount`:
+
+```go
+if err := keeper.icaControllerKeeper.RegisterInterchainAccount(ctx, connectionID, owner.String(), version); err != nil {
+    return err
+}
+
+return nil
+```
+
+The `version` argument is used to support ICS29 fee middleware for relayer incentivization of ICS-27 packets. Consumers of the `RegisterInterchainAccount` are expected to build the appropriate JSON encoded version string themselves and pass it accordingly. If an empty string is passed in the `version` argument, then the version will be initialized to a default value in the `OnChanOpenInit` callback of the controller's handler, so that channel handshake can proceed.
```

Comment:
```suggestion
The `version` argument is used to support ICS-29 fee middleware for relayer incentivization of ICS-27 packets. Consumers of the `RegisterInterchainAccount` are expected to build the appropriate JSON encoded version string themselves and pass it accordingly. If an empty string is passed in the `version` argument, then the version will be initialized to a default value in the `OnChanOpenInit` callback of the controller's handler, so that channel handshake can proceed.
```
For consistency with other uses